### PR TITLE
Change a sizable portion of LocationXY(Z) to (Tile)CoordsXY(Z)

### PR DESCRIPTION
**Pull request by SijmenSchoon**
This is the first working state from changing the type of `paint_session.MapPosition` to `CoordsXY`. One small step towards fixing #9943 

I double checked to not mix up any `Coords` and `TileCoords` variables, but I might have missed some.

### CODE REVIEW

File: src/openrct2/paint/Supports.cpp [hunk 1]
```diff
@@ -16,16 +16,16 @@
...
-    {4,  4},
-    {28, 4},
-    {4,  28},
-    {28, 28},
-    {16, 16},
-    {16, 4},
-    {4,  16},
-    {28, 16},
-    {16, 28}
+static constexpr const CoordsXY SupportBoundBoxes[] = {
```
Comment (tupaschoal) [inline src/openrct2/paint/Supports.cpp]: You could have changed only the type of the `constexpr` to `CoordsXY`, it would know how to initialize itself with the initializer list above (see [example](https://godbolt.org/z/NvYi1w)). This applies to many other places in the patch.

Also if you hadn't changed the variable name, it would make it easier to review, as there would be less formatting changes due to the new width ;P

Comment (SijmenSchoon) [reply src/openrct2/paint/Supports.cpp]: I know, but I found using the constructors prettier. I can change it back if others agree though

Comment (duncanspumpkin) [reply src/openrct2/paint/Supports.cpp]: I don't think we really do it like this anywhere else in the codebase. I think it would look neater keeping the old format.

Comment (SijmenSchoon) [reply src/openrct2/paint/Supports.cpp]: Alright, I will change it back tonight

File: src/openrct2/paint/Supports.cpp [hunk 2]
```diff
@@ -772,8 +772,8 @@ bool metal_a_supports_paint_setup(
     }
     else
     {
-        int8_t xOffset = loc_97AF20[segment].x;
-        int8_t yOffset = loc_97AF20[segment].y;
+        int8_t xOffset = SupportBoundBoxes[segment].x;
```
Comment (ZehMatt) [inline src/openrct2/paint/Supports.cpp]: We can avoid this assignment if we refactor sub_98196C to also use CoordsXY, but this should be done in another PR. Right now it accesses the memory 3 times which isn't too nice.

Comment (SijmenSchoon) [reply src/openrct2/paint/Supports.cpp]: What do you suggest I do right now?

Comment (ZehMatt) [reply src/openrct2/paint/Supports.cpp]: Nothing, like I stated this should be done in another PR honestly as it will change the declaration of those functions and requires a lot more modifications.

File: src/openrct2/ride/TrackPaint.cpp [hunk 3]
```diff
@@ -29,15 +29,15 @@
 
 // clang-format off
 /* rct2: 0x007667AC */
-static LocationXY16 loc_7667AC[] = {
+static TileCoordsXY EntranceOffsetEdgeNE[] = {
```
Comment (ZehMatt) [inline src/openrct2/ride/TrackPaint.cpp]: Can use constexpr here.

File: src/openrct2/ride/TrackPaint.cpp [hunk 4]
```diff
@@ -29,15 +29,15 @@
...
     { -1, 0 },
     { 0,  -1 },
     { 1,  0 },
     { 0,  1 },
 };
 
 /* rct2: 0x007667AE */
-static LocationXY16 loc_7667AE[] = {
+static TileCoordsXY EntranceOffsetEdgeNW[] = {
```
Comment (ZehMatt) [inline src/openrct2/ride/TrackPaint.cpp]: Can also use constexpr.

File: src/openrct2/paint/Supports.cpp (hunk 2 above)
Comment (SijmenSchoon) [reply src/openrct2/paint/Supports.cpp]: Alright, cool

File: src/openrct2/ride/TrackPaint.cpp (hunk 3 above)
Comment (SijmenSchoon) [reply src/openrct2/ride/TrackPaint.cpp]: Can't currently, since `TileCoordsXY(int32_t, int32_t)` is not `constexpr` (unlike `TileCoordsXYZ(int32_t, int32_t, int32_t)`. I suppose this is an error, I'll change that too

Comment (SijmenSchoon) [reply src/openrct2/ride/TrackPaint.cpp]: There's a bunch of `const CoordsXY(Z)` declarations in this file - should I also change these to `constexpr`s?

Comment (SijmenSchoon) [reply src/openrct2/ride/TrackPaint.cpp]: There's a bunch of `const CoordsXY(Z)`s in this file, should I change these to `constexpr`s too?

**Comment by SijmenSchoon** [comment]
@tupaschoal I wanted to request a review, but the button does not work

**Comment by duncanspumpkin** [comment]
I suspect you can only request review from team members it's the same for assigning issues.
